helper/testhelpers/testhelpers.go

```diff
@@ -767,7 +766,7 @@ func RaftClusterJoinNodes(t testing.T, cluster *vault.TestCluster) {
 	{
 		core := cluster.Cores[1]
 		core.UnderlyingRawStorage.(*raft.RaftBackend).SetServerAddressProvider(addressProvider)
-		_, err := core.JoinRaftCluster(namespace.RootContext(context.Background()), leaderAPI, leaderCore.TLSConfig, false)
+		_, err := core.JoinRaftCluster(namespace.RootContext(context.Background()), leaderCore.Client, false)
```

Comment:
I don't think we want to use this client here as it does retries, which is not ideal for Joining

Reply:
Fixed.